### PR TITLE
docs: Added table of contents to contributor docs

### LINKED ISSUE #856: Improve CONTRIBUTING.md file by adding a table of contents
### Is this a bug or feature request?

doc feature  

### What is the current behavior?

The [CONTRIBUTING.md](https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md) file has no table of contents for easy access to sub-sections.

### What is the expected or desired behavior?

To include a table of contents at the top of the file in the spirit of [this example](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) from Rust's contributing docs. I think this will be useful because CONTRIBUTING.md is a bit long and it'll give contributors a way to visualize all sections of the file in one glance, and a place to access them quickly.

### PR DESCRIPTION
**Pull request by gilbertginsberg**
Fixes #856

### CODE REVIEW

**Comment by kumar303** [comment]
Awesome! A TOC will be really helpful. I'm worried about people forgetting to update it though. Can we use a script instead, maybe this one? https://github.com/thlorenz/doctoc

**Comment by gilbertginsberg** [comment]
Oh nice. Sure, I'll look into implementing @kumar303.

**Comment by gilbertginsberg** [comment]
@kumar303: TOC is now generated with doctoc. Nifty tool.

**Comment by gilbertginsberg** [comment]
One thing to note: whoever updates `CONTRIBUTING.md` will still need to remember to update the TOC by running the command `doctoc CONTRIBUTING.md`.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,9 +2,36 @@
 
 Hi! Thanks for your interest in helping make
 [WebExtension](https://developer.mozilla.org/en-US/Add-ons/WebExtensions)
-development more awesome by contributing to the `web-ext` tool.
-
-#Picking an issue
+development more awesome by contributing to the `web-ext` tool. Here are links to all the sections in this document:
+
```
Comment (kumar303) [inline CONTRIBUTING.md]: You should put a comment here that says how to run `doctoc`. Also, you'll need to add it to `package.json` as a dev requirement and make a script we can use like `npm run gen-contributing-toc`.

Comment (gilbertginsberg) [inline CONTRIBUTING.md]: Okay, thanks @kumar303. I'll get to work on this.

Comment (gilbertginsberg) [inline CONTRIBUTING.md]: @kumar303: I've made these three changes. Will be alert if others are needed.

**Comment by gilbertginsberg** [comment]
Good call re: the note, @kumar303 . I wondered about this.

I debated what the best spot for it would be, and looked around for a repo with something similar. And ultimately went with a new Documentation heading because it seems natural there and perhaps can be of use down the road. I thought about including it in the pull request or commit sections since updating the TOC is part of the workflow. But then decided a note about CONTRIBUTING.md in those spots might be a bit of a non sequitur...

**Comment by kumar303** [comment]
@rpl can you think of a fancy way to automatically run the TOC generator script? I can't think of anything that wouldn't have some annoyance to it. It's probably ok to run it manually since editing contributing.md is a rare occurrence.

**Comment by kumar303** [comment]
@gilbertginsberg whoops, I almost lost track of this. Can you resolve the conflicts? The toc might need to be updated. After that, I think we can merge it in and just enforce toc generation by word of mouth for now.

**Comment by gilbertginsberg** [comment]
@kumar303: Sure thing, just resolved the conflicts.

**Pull request merged**